examples/reinforcement_learning/dqn_cartpole.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python
+"""Example code of DQN and DoubleDQN on OpenAI Gym environments.
+
+For DQN, see: http://www.nature.com/articles/nature14236
+For DoubleDQN, see: https://arxiv.org/abs/1509.06461
+"""
+from __future__ import print_function
+from __future__ import division
+import argparse
+import collections
+import copy
+import random
+
+import gym
+import numpy as np
+
+import chainer
+from chainer import functions as F
+from chainer import links as L
+from chainer import optimizers
+
+
+class QFunction(chainer.Chain):
+    """Q-function represented by a MLP."""
+
+    def __init__(self, ndim_obs, n_actions, n_units=100):
+        super(QFunction, self).__init__(
+            l0=L.Linear(ndim_obs, n_units),
+            l1=L.Linear(n_units, n_units),
+            l2=L.Linear(n_units, n_actions),
+        )
+
+    def __call__(self, x):
+        """Compute Q-values of actions for given observations."""
+        h = F.relu(self.l0(x))
+        h = F.relu(self.l1(h))
+        return self.l2(h)
+
+
+def get_greedy_action(Q, obs):
+    xp = Q.xp
+    obs = xp.expand_dims(xp.asarray(obs, dtype=np.float32), 0)
+    with chainer.no_backprop_mode():
+        q = Q(obs).data[0]
+    return int(xp.argmax(q))
+
+
+def mean_clipped_loss(y, t):
+    # Add an axis because F.huber_loss only accepts arrays with ndim >= 2
+    y = F.expand_dims(y, axis=-1)
+    t = F.expand_dims(t, axis=-1)
+    return F.sum(F.huber_loss(y, t, 1.0)) / y.shape[0]
+
+
+def update(Q, target_Q, opt, samples, gamma=0.99, target_type='double_dqn'):
+    xp = Q.xp
+    s = xp.asarray([sample[0] for sample in samples], dtype=np.float32)
```

Comment:
Ditto. Can you rename the variables:
s -> state
a -> action
r -> reward
s_next -> state_next

Alternatively, if the letters are very common in RL literature, could you describe in the beginning of update() that samples is a list of tuples of (state, action... )?